File: ZezeJava/ZezeJava/src/main/java/Zeze/Util/Random.java
package Zeze.Util;

import java.util.List;
import java.util.concurrent.ThreadLocalRandom;
import Zeze.Net.Binary;
import org.jetbrains.annotations.NotNull;

public final class Random {
	public static @NotNull ThreadLocalRandom getInstance() {
		return ThreadLocalRandom.current();
	}

	public static byte @NotNull [] nextBytes(int size) {
		var bytes = new byte[size];
		getInstance().nextBytes(bytes);
		return bytes;
	}

	public static @NotNull Binary nextBinary(int size) {
		return new Binary(nextBytes(size));
	}

	public static <T> @NotNull List<T> shuffle(@NotNull List<T> list) {
		var random = getInstance();
		for (int i = 1, n = list.size(); i < n; i++) {
			int pos = random.nextInt(i + 1);
			var x = list.get(i);
			list.set(i, list.get(pos));
			list.set(pos, x);
		}
		return list;
	}

	public static <T> T @NotNull [] shuffle(T @NotNull [] list) {
		var random = getInstance();
		for (int i = 1, n = list.length; i < n; i++) {
			int pos = random.nextInt(i + 1);
			var x = list[i];
			list[i] = list[pos];
			list[pos] = x;
		}
		return list;
	}

	public static int @NotNull [] shuffle(int @NotNull [] list) {
		var random = getInstance();
		for (int i = 1, n = list.length; i < n; i++) {
			int pos = random.nextInt(i + 1);
			var x = list[i];
			list[i] = list[pos];
			list[pos] = x;
		}
		return list;
	}

	public static long @NotNull [] shuffle(long @NotNull [] list) {
		var random = getInstance();
		for (int i = 1, n = list.length; i < n; i++) {
			int pos = random.nextInt(i + 1);
			var x = list[i];
			list[i] = list[pos];
			list[pos] = x;
		}
		return list;
	}
}

File: projects/zezeboot/client_cs/Zeze/Util/Random.cs
using System.Collections.Generic;

namespace Zeze.Util
{
    public sealed class Random
    {
        public static System.Random Instance { get; } = new System.Random();

        public static IList<T> Shuffle<T>(IList<T> list)
        {
            for (int i = 1; i < list.Count; i++)
            {
                int pos = Instance.Next(i + 1);
                (list[i], list[pos]) = (list[pos], list[i]);
            }
            return list;
        }

        public static T[] Shuffle<T>(T[] list)
        {
            for (int i = 1; i < list.Length; i++)
            {
                int pos = Instance.Next(i + 1);
                (list[i], list[pos]) = (list[pos], list[i]);
            }
            return list;
        }
    }
}
